src/cargo/ops/cargo_add/mod.rs

```diff
@@ -733,7 +749,12 @@ fn select_package(
                     anyhow::bail!("the crate `{dependency}` could not be found at `{source}`")
                 }
                 1 => {
-                    let mut dep = Dependency::from(&possibilities[0]);
+                    let base_name_and_value = dependency
+                        .source()
+                        .map(|s| s.as_path().map(|p| &p.base_name_and_value))
+                        .flatten()
+                        .unwrap_or(&None);
+                    let mut dep = Dependency::from_summary(&possibilities[0], base_name_and_value);
                     if let Some(reg_name) = dependency.registry.as_deref() {
```

Comment:
Like we d with `registry`, I think I'd rather handle it here than in `from_summary`.  `from_summary` is constructing it according to the `Summary`.  We then manually manipulate it for our needs.  Trying to mix the two is mixing of concerns and inconsistently doing so since we have the registry code here.